src/main/java/brooklyn/entity/proxy/aws/ElbControllerImpl.java

```diff
@@ -579,15 +689,101 @@ protected void deleteLoadBalancer(String elbName) {
+    private void removeSecurityGroups() {
+        Collection<String> securityGroups = sensors().get(LOAD_BALANCER_SECURITY_GROUPS);
+
+        final AmazonEC2Client client = ec2Client(getLocation());
+        if (securityGroups != null) {
+            List<Task> removals = MutableList.of();
+            for (String groupId : securityGroups) {
+                removals.add(removeOneSecurityGroup(client, groupId));
+            }
+            ParallelTask<Boolean> invoke = new ParallelTask<>(MutableMap.of(
+                "displayName", "Delete security groups",
+                "description", "Deleting " + removals,
+                "tag", BrooklynTaskTags.tagForCallerEntity(this)
+            ), removals);
+            TaskTags.markInessential(invoke);
+            DynamicTasks.queueIfPossible(invoke).orSubmitAsync(this).andWaitForSuccess();
+        }
+    }
+
+    private Task<Boolean> removeOneSecurityGroup(final AmazonEC2Client client, final String groupId) {
+        return Tasks.create(groupId, new Callable<Boolean>() {
+
+            @Override
+            public Boolean call() throws Exception {
+                ReferenceWithError<Boolean> result;
+                final Repeater repeater = Repeater.create("Delete " + groupId)
+                    .limitTimeTo(2L, TimeUnit.MINUTES)
+                    .every(10, TimeUnit.SECONDS)
+                    .until(new DeleteASecurityGroup(client, groupId))
+                    .rethrowException();
+
+                Tasks.setBlockingDetails(repeater.getDescription());
+                try {
+                    result = repeater.runKeepingError();
+                } finally {
+                    Tasks.resetBlockingDetails();
+                }
+
+                if (Boolean.TRUE.equals(result.getWithoutError()))
+                    return true;
+                if (result.hasError())
+                    throw Exceptions.propagate(result.getError());
+                return false;
+            }
+        });
+    }
+
+    private static class DeleteASecurityGroup implements Callable<Boolean> {
+        private AmazonEC2Client client;
+        private String groupId;
+        public DeleteASecurityGroup(final AmazonEC2Client client, final String groupId) {
+            this.client = client;
+            this.groupId = groupId;
+        }
+
+        /**
+         * Delete the group. If it is still in use, return false to indicate that a retry should be attempted.
+         * @return true unless a retry should be attempted because the group is still in use.
+         */
+        @Override
+        public Boolean call() throws Exception {
+            final DeleteSecurityGroupRequest deleteGroup = new DeleteSecurityGroupRequest().withGroupId(groupId);
+            boolean success = true;
+            try {
+                LOG.debug("Removing security group {}", groupId);
+                client.deleteSecurityGroup(deleteGroup);
+                LOG.debug("Removed security group {}", groupId);
```

Comment:
Handy to log `this` as well so it's clear in the log which entity the message came from.